src/context/core.rs

```diff
@@ -38,6 +42,8 @@ pub trait Context: std::fmt::Debug + std::fmt::Display {
         self.env().eval(expr)
     }
 
+    fn eval_mutable(&mut self, expr: Expr) -> EvalResult;
```

Comment:
What do you think about defaulting to throwing an error? Just for the sake of being explicit, I think this might help to be more intentional about which things should be able to be accessed mutably. 

```suggestion
    fn eval_mutable(&mut self, expr: Expr) -> EvalResult {
      Err(Error::CannotEvaluateAsMutable(expr.clone()));
    }
```

Reply:
Yes, this makes sense. Only I feel that this is a development error and users should eventually never experience this, so I feel throwing an `internal_err!()` is sufficient, but maybe I am missing something

Reply:
forget this, your next comment made me understand the point